### ISSUE TITLE
Upgrading from 2.2 SDK to 3.0 introduces NETSDK1047 error

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/sdk#2620]
We're trying to upgrade the master branch of aspnetcore to build with the latest 3.0.100-alpha builds, and running into this error which didn't happen in the 2.2 SDK.

> C:\b\w\695426870b618c58\.dotnet\x64\sdk\3.0.100-alpha1-009701\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(208,5): error NETSDK1047: Assets file 'C:\b\w\695426870b618c58\modules\KestrelHttpServer\test\Kestrel.Core.Tests\obj\project.assets.json' doesn't have a target for '.NETCoreApp,Version=v2.2/win-x64'. Ensure that restore has run and that you have included 'netcoreapp2.2' in the TargetFrameworks for your project. You may also need to include 'win-x64' in your project's RuntimeIdentifiers. [C:\b\w\695426870b618c58\modules\KestrelHttpServer\test\Kestrel.Core.Tests\Kestrel.Core.Tests.csproj]

I can't figure out why I'm getting an error about `win-x64` because we don't set this RID in our project files. I can provide a binlog if you want.

**To reproduce:**
1. Clone https://github.com/aspnet/KestrelHttpServer/tree/a8073167e7adcca0d6708d4da56e43066535e3a5
1. Using the 3.0.100-alpha1-009701 SDK, try to build the KestrelHttpServer.sln.

**Comment by livarcocc** [comment]
@peterhuene any chance this is happening because of our changes to include a host by default?

**Comment by peterhuene** [comment]
More than likely.  A binlog would be useful as to why the restore didn't take place for `win-x64` to get the apphost in the assets file (note: this is temporary functionality until we can get "download-only" nuget packages).

**Comment by natemcmaster** [comment]
[KestrelHttpServer.binlog.zip](https://github.com/dotnet/sdk/files/2512591/KestrelHttpServer.binlog.zip)

**Comment by peterhuene** [comment]
It appears setting `HasRuntimeOutput` explicitly to true is the cause, which is done here:

https://github.com/aspnet/KestrelHttpServer/blob/a8073167e7adcca0d6708d4da56e43066535e3a5/test/Directory.Build.props#L16

There appears to be a discrepancy between our defaulting of `UseAppHost` based on `HasRuntimeOutput` and `DefaultAppHostRuntimeIdentifier` on `_IsExecutable`.

This discrepancy is causing the restore to take place with an empty `DefaultAppHostRuntimeIdentifier`, but the build has a non-empty identifier, which causes the failure to find the asset in the graph.

I'm working on a fix.

**Comment by natemcmaster** [comment]
Thanks for getting a fix in so quickly @peterhuene. How long does it take for this change to flow up into core-sdk? We tried updating to 3.0.100 with yesterday's latest build (3.0.100-alpha1-009708), and still hit NETSDK1047 errors.

**Comment by livarcocc** [comment]
Not yet. That will happen once we finish the arcade move.

**Comment by peterhuene** [comment]
Ok, so we need an old school insertion into CLI for the most recent SDK then, I assume.  I'll make that happen today.

**Comment by natemcmaster** [comment]
Thanks!

**Comment by natemcmaster** [comment]
I tried again to update to the latest build listed on dotnet/core-sdk, version 3.0.100-preview-009728, built from https://github.com/dotnet/core-sdk/tree/ac9f27d723ca5c5503a8faa5c11e60fdecf46dc7. I'm still seeing this error. Any ETA on getting the fix into a build of the CLI?

**Comment by peterhuene** [comment]
`dotnet/sdk` changes haven't flowed through to `core-sdk` in quite some time.

I have a PR to update it in `dotnet/cli` (https://github.com/dotnet/cli/pull/10264).  When that is merged, I need to update the toolset, and then finally `core-sdk`.  This is a bit of a mess due to the sdk repos moving to arcade and having no orchestration set up yet for 3.0.

I'll follow up once a build contains the fix.

**Comment by natemcmaster** [comment]
Thanks @peterhuene

**Comment by peterhuene** [comment]
@natemcmaster The fix should now be in `3.0.100-preview-009730`.  Thanks!

**Comment by Tragetaschen** [comment]
I'm running with `3.0.100-preview-009756` and still get a very similar error (though it's in German)
```
C:\Program Files\dotnet\sdk\3.0.100-preview-009756\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(220,5): error NETSDK1047: Die Ressourcendatei "…\obj\project.assets.json" verfügt über kein Ziel für ".NETCoreApp,Version=v2.1/win-x64". Stellen Sie sicher, dass die Wiederherstellung ausgeführt wurde, und dass Sie "netcoreapp2.1" in die TargetFrameworks für Ihr Projekt aufgenommen haben. Unter Umständen müssen Sie auch "win-x64" in die RuntimeIdentifiers Ihres Projekts aufnehmen. [….csproj]
```
The project with the error is a library for the project I'm building and targeting
`<TargetFrameworks>netstandard2.0;netcoreapp2.1</TargetFrameworks>`, the project I'm building only targets `netcoreapp2.1`.
